httpx/middleware/header.go

```diff
@@ -0,0 +1,47 @@
+package middleware
+
+import (
+	"net/http"
+
+	"github.com/remind101/pkg/httpx"
+	"golang.org/x/net/context"
+)
+
+type Header struct {
```

Comment:
It would be nice to add some documentation here to explain the purpose of this middleware. I can see its meant to grab a header from the request and put it in the context. Is it meant to be a generalization of the request id middleware? If so, should that middleware be rewritten to use this one?

Reply:
Yeah, that's right. I'll update the request one.

The real world usage  is:
https://github.com/remind101/r101-integrations/blob/master/cmd/app/main.go#L32
https://github.com/remind101/r101-integrations/blob/master/cmd/app/main.go#L109
https://github.com/remind101/r101-integrations/blob/master/cmd/app/main.go#L135

I suspect this will be the primary use case, but I didn't want to put our private headers in an open source repo.